### PR TITLE
[HIBIKE] Add working RFID firmware

### PR DESCRIPTION
**Pull request by onibrow**
Functionality: Prints out the last byte of the UID of an RFID tag, along with whether or not a tag is detected.

[uint8_t id, uint8_t tag_detect] where id is the last byte, and tag_detect is a 1 or 0 depending if a tag is present.

### CODE REVIEW

File: hibike/README.md [hunk 1]
```diff
@@ -211,6 +211,8 @@ Device Type Enumeration:
 |         |                | 12           | enc_pos             | float      | yes   | yes    |
 |         |                | 13           | enc_vel             | float      | yes   | no     |
 |         |                | 14           | motor_current       | float      | yes   | no     |
+|  0x0B   | RFID           | 0            | id      | uint8_t | yes | no |
+|         |                | 1            | detect_tag | bool       | yes   | no |
```
Comment (dHutchings) [inline hibike/README.md]: Tag detect is declared to be a bool in the README.  It's a uint8_t in the json and the code.

Please use only one type.  (I'd prefer bool)

File: hibike/devices/RFID/rfid.cpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+// hibike_loop will call device_write and device_read
+
+void loop() {
+  // do whatever you want here
+  // note that hibike will only process one packet per call to hibike_loop()
+  // so exessive delays here will affect hibike.
+  // Look for new cards
+  hibike_loop();
+  if ( ! mfrc522.PICC_IsNewCardPresent()) {
+    if (del) {
```
Comment (dHutchings) [inline hibike/devices/RFID/rfid.cpp]: Please, at least one comment in this whole loop function.

What we discovered about the consecutive reads should (and hence why we need del) should be recorded somewhere.

File: hibike/devices/RFID/rfid.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#ifndef EX_DEVICE_H
```
Comment (dHutchings) [inline hibike/devices/RFID/rfid.h]: This device is not an example device.  Please rename to 

#ifndef RFID_H

File: hibike/devices/RFID/rfid.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  1,                      // Year
+  UID_RANDOM,     // ID
+};
+///////////////////////////////////////////////
+
+// this is an optional device specific enum to make implementation easier
+typedef enum {
+  ID = 0,
+  TAG_DETECT = 1,
+} ExampleDeviceParam;
```
Comment (dHutchings) [inline hibike/devices/RFID/rfid.h]: Think you can "Example Device Param" to just plain "param".  let's try to scrub any remaining references to "Example"

**Review by dHutchings — CHANGES_REQUESTED** [review]
Again, good job getting this working!

Just a few minor changes.  Once they're done, i'll approve the PR.

Not going to code review the examples, since they're just examples.

**Comment by onibrow** [comment]
The newest changes have been pushed. This removes all references to example device, and cleans up code while adding some comments.

File: hibike/README.md (hunk 1 above)
Comment (dHutchings) [reply hibike/README.md]: fixed.

**Review by dHutchings — APPROVED** [review]

**Comment by dHutchings** [comment]
@onibrow   Approved.  Squashing & merging.  Good job.

I'm deleting hibike/rfid from github.  Please remove from your local machine.

**Pull request merged**